packages/blockchain-wallet-v4-frontend/src/modals/Swap/PreviewSwap/PreviewSwapSuccess.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useEffect, useState } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { useDispatch } from 'react-redux'
+import { Form } from 'redux-form'
+import styled from 'styled-components'
+
+import { formatCoin } from '@core/exchange/utils'
+import Remote from '@core/remote'
+import { PaymentValue, SwapNewQuoteStateType } from '@core/types'
+import { Button, HeartbeatLoader, Icon, Link, Text, TextGroup } from 'blockchain-info-components'
+import { ErrorCartridge } from 'components/Cartridge'
+import { FlyoutWrapper, Row, Title, Value } from 'components/Flyout'
+import { GenericNabuErrorFlyout } from 'components/GenericNabuErrorFlyout'
+import { actions } from 'data'
+import { Analytics, IncomingAmount, InitSwapFormValuesType, SwapAmountFormValues } from 'data/types'
+import { isNabuError } from 'services/errors'
+
+import { QuoteCountDown } from '../../components/QuoteCountDown'
+import { FeeBreakdownBox, FromToLogoLeft, TopText } from '../components'
+
+const StyledRow = styled(Row)`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: center;
+  padding-top: 30px;
+  padding-bottom: 24px;
+
+  & > div:first-child {
+    width: 100%;
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  }
+`
+const ToolTipText = styled.div`
+  display: flex;
+  flex-direction: row;
+  border-radius: 8px;
+  margin-top: 8px;
+  padding: 16px;
+  background-color: ${(props) => props.theme.grey000};
+
+  animation: fadeIn 0.3s ease-in-out;
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`
+
+const ExchangeRateRow = styled.div`
+  display: flex;
+`
+const IconWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  cursor: pointer;
+  margin-left: 4px;
+`
+
+const QuoteCountDownWrapper = styled.div`
+  margin-top: 28px;
+`
+
+type Props = {
+  error?: any
+  incomingAmount: IncomingAmount
+  initSwapFormValues: InitSwapFormValuesType
+  onClickBack: () => void
+  payment: PaymentValue | undefined
+  quote: SwapNewQuoteStateType
+  submitting: boolean
+  swapAmountFormValues: SwapAmountFormValues
+}
+
+export const PreviewSwapSuccess = (props: Props) => {
+  const [isActiveExchangeToolTip, setIsActiveExchangeToolTip] = useState(false)
+  const dispatch = useDispatch()
```

Comment:
Can we get rid of dispatch here, and make this component rely on callbacks from props?